apps/language_server/test/source_file_test.exs

```diff
@@ -875,4 +879,44 @@ defmodule ElixirLS.LanguageServer.SourceFileTest do
       end
     end
   end
+
+  describe "lsp_character_to_elixir" do
+    test "it should handle a nil line" do
+      assert 0 = SourceFile.line_length_utf16(nil)
+    end
+  end
+
+  describe "formatter_for" do
+    setup [:temporary_elixir_file]
+
+    def temporary_elixir_file(_) do
+      unique_number = System.unique_integer([:positive, :monotonic])
+      path = Path.join([System.tmp_dir!(), "formatted_#{unique_number}.ex"])
+      File.touch!(path)
+      uri = %URI{path: path, scheme: "file://"}
+
+      on_exit(fn ->
+        File.rm(path)
+      end)
+
+      {:ok, file_path: path, file_uri: URI.to_string(uri)}
+    end
+
+    def as_contents(contents, %{file_path: file_path}) do
+      File.write!(file_path, contents)
+    end
+
+    test "it should not crash on syntax errors", ctx do
+      as_contents("", ctx)
+
+      patch(Mix.Tasks.Format, :formatter_for_file, fn _ -> raise %SyntaxError{} end)
```

Comment:
Let's not bring in another dependency just to produce a syntax error. BTW according to https://hexdocs.pm/patch/readme.html#limitations Patch is not compatible with async: true.

Reply:
Producing a sytax error here is pretty tough; patch is an awesome library, and I think you should reconsider.